### ISSUE TITLE
Unknown ioctl(0xc0404807): type:0x48 nr:0x7 dir:0x3 size:64

### ISSUE THREAD

**Issue by db48x** [rr-debugger/rr#3804]
```
 -> Assertion `t->regs().syscall_result_signed() == -syscall_state.expect_errno' failed to hold. Expected EINVAL for 'ioctl' but got result 0 (errno SUCCESS); Unknown ioctl(0x90044802): type:0x48 nr:0x2 dir:0x2 size:4100 addr:0x7fffb3e6dab0
```

later:

```
 -> Assertion `t->regs().syscall_result_signed() == -syscall_state.expect_errno' failed to hold. Expected EINVAL for 'ioctl' but got result -32 (errno EPIPE); Unknown ioctl(0xc0404807): type:0x48 nr:0x7 dir:0x3 size:64 addr:0x7ffd1e75b060
```

I tried to add `HIDIOCGREPORT` to `prepare_ioctl`, but I got a little lost in the twisty passages. On the other hand, I think that I was able to add `HIDIOCGRDESC` correctly.

**Comment by db48x** [comment]
Oh, that’s because HIDIOCGRDESC isn’t even the right ioctl.

**Comment by db48x** [comment]
Well I think I did it correctly; let me send you the diff and see what you say.

**Comment by rocallahan** [comment]
I didn't mean to merge support for the unknown ioctl, so I've reverted that part. If it still matters, reopen this issue or open another issue and we can investigate further. The first step would be to figure out what file it's being called on.

**Comment by db48x** [comment]
Yea, I spent an hour or two trying to track this down over the weekend but didn’t get very far. Did you know that GDB will reject your catchpoint if the syscall number is negative? Super dumb. And the kernel’s built–in ftrace stuff only supports known syscalls; it doesn't have a catchall that traces all syscalls and lets you filter them by the syscall number. Also really dumb. I had decided that I should strace the thing and just log every file it opens. Most of them will be irrelevant, but maybe the relevant one will be obvious. I ran out of motivation this weekend, but I’m not doing anything important right now so I’ll let you know if I find anything.
